### ISSUE TITLE
WCAG: Mark fields as required when in a table cell

### ISSUE THREAD

**Issue by Febakke** [Altinn/app-frontend-react#1664]
### Description of the bug

This issue is about how the repeating group behaves in the version described here: https://docs.altinn.studio/app/development/ux/fields/grouping/repeating/table/#showing-components-directly-in-the-table

1. [ ] When you use the repeating group as a table. (The version where the user can fill out fields within the table) there are no indicator telling the user what fields are required. This is also true with grid.

2. [x] ~~When there is a an error on one of the rows the whole background of the row is colored red. This is used normaly to indicate that there are an error inside the row on a normal repeating group, but it does not work well in this version. We should remove the background color when is in this version. The wcag issue is that the contrast is low on the error message.~~

**Example:**
<img width="546" alt="Skjermbilde 2023-11-23 kl  09 40 09" src="https://github.com/Altinn/app-frontend-react/assets/33222679/1618e380-95f7-4876-99d7-857a2ec71e95">

### Steps To Reproduce

1. Go to an app with a repeating group in table mode that contains required fields
2. Try to go the page beyond the group
3. Look at the errors in the table and try to find the indicator telling the user that it is required

**Comment by lassopicasso** [comment]
Related grid issue: https://altinn.slack.com/archives/C02FK32KVKJ/p1701164474563379

**Comment by olavflar** [comment]
<img width="328" alt="Image" src="https://github.com/user-attachments/assets/c75cc4ee-c70c-4cee-be75-f001d2991697" />
Looks like this is solved?

**Comment by Febakke** [comment]
> <img alt="Image" width="328" src="https://private-user-images.githubusercontent.com/70377508/397295938-c75cc4ee-c70c-4cee-be75-f001d2991697.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ1OTg1MTUsIm5iZiI6MTczNDU5ODIxNSwicGF0aCI6Ii83MDM3NzUwOC8zOTcyOTU5MzgtYzc1Y2M0ZWUtYzcwYy00Y2VlLWJlNzUtZjAwMWQyOTkxNjk3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE5VDA4NTAxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM5NTY0YTdiODhhZmQ0ZTE0MmEwZmJmMTVkMjQ0OGQwZmYzOWU3ZGQ3MzhkMjc1YWQ5NGRiM2E3OTkyZTFkN2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xE2BNdJnKN79yd826phhvW-98PHtNbdTV6YyRXgxUHI"> Looks like this is solved?

I get alerts for this as i made the issue back in the days 👴 And i cant help, get involvled! 
I would double check the contrast on delete button if you havent already. I find it a little hard to read. I remember we talked about removing the background color all together as it is a little excessive compared to other error messages. Would also mitigate contrast issues.

**Comment by bjosttveit** [comment]
If we are strictly talking about repeating groups with components directly in the rows then I tried fixing it a little while ago, and looks like this now:

<img width="997" alt="Image" src="https://github.com/user-attachments/assets/a20c4582-2c0f-4671-a3e4-3adc1789744c" />

However, I may have just found a bug with it, it seems to highlight the background again if there are errors on the repeating group component itself (in this case, minCount), this should not cause rows to get highlighted...

<img width="997" alt="Image" src="https://github.com/user-attachments/assets/4d880f2b-4785-4fb2-adce-04713268cb1c" />

**Comment by olemartinorg** [comment]
I added a new issue about the delete button contrast (and validation message contrast as well, I guess) in #2892. Edited the title here to be about the remaining issue about a required indicator when components are shown in table cells.
